homeassistant/components/bluetooth_tracker/device_tracker.py

```diff
@@ -168,24 +183,102 @@ def lookup_name(mac: str) -> Optional[str]:
         # If an update is in progress, we don't do anything
         if update_bluetooth_lock.locked():
             _LOGGER.debug(
-                "Previous execution of update_bluetooth is taking longer than the scheduled update of interval %s",
+                "Previous execution of update_bluetooth "
+                "is taking longer than the scheduled update of interval %s",
                 interval,
             )
             return
 
         async with update_bluetooth_lock:
             await perform_bluetooth_update()
 
-    async def handle_manual_update_bluetooth(call):
-        """Update bluetooth devices on demand."""
-
-        await update_bluetooth()
-
     hass.async_create_task(update_bluetooth())
     async_track_time_interval(hass, update_bluetooth, interval)
 
-    hass.services.async_register(
-        DOMAIN, "bluetooth_tracker_update", handle_manual_update_bluetooth
-    )
 
-    return True
+class BluetoothEntity(ScannerEntity):
+    """Represent a tracked device that is on a scanned bluetooth network."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, mac: str, device_name: str, attributes: dict
+    ):
+        """Set up bluetooth entity."""
+        self._attributes = attributes
+        self._config_entry = config_entry
+        self._last_seen = dt_util.utcnow()
+        self._mac = mac
+        self._name = device_name
+        self._unsub_dispatcher = None
+
+    def __hash__(self):
+        """Return the hash for the entity."""
+        return hash(self.mac)
+
+    def __eq__(self, other):
+        """Check whether this entity is equal to another."""
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self.mac == other.mac
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {"name": self.name, "identifiers": {(DOMAIN, self.mac)}}
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific attributes."""
+        return self._attributes
+
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        if (
+            dt_util.utcnow() - self._last_seen
+            < self._config_entry.options[CONF_CONSIDER_HOME]
+        ):
+            return True
+
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def mac(self):
```

Comment:
Remove the property and make mac a non private instance attribute if we need to access it from outside.